presto-main/src/main/java/io/prestosql/operator/DynamicFilterSourceOperator.java

```diff
@@ -208,13 +236,61 @@ public void addInput(Page page)
 
     private void handleTooLargePredicate()
     {
-        // The resulting predicate is too large, allow all probe-side values to be read.
-        dynamicPredicateConsumer.accept(TupleDomain.all());
+        // The resulting predicate is too large, convert to min/max per column for orderable types
+        if (minMaxValues) {
+            for (int channelIndex = 0; channelIndex < channels.size(); ++channelIndex) {
+                Type type = channels.get(channelIndex).type;
+                if (type.isOrderable()) {
+                    Block block = blockBuilders[channelIndex].build();
+                    updateMinMaxValues(block, type, channelIndex);
+                }
+            }
+        }
         // Drop references to collected values.
         valueSets = null;
         blockBuilders = null;
     }
 
+    private void updateMinMaxValues(Block block, Type type, int channelIndex)
+    {
+        int minValuePosition = -1;
+        int maxValuePosition = -1;
+
+        for (int position = 0; position < block.getPositionCount(); ++position) {
```

Comment:
I guess that this loop can benefit from JIT compilation, especially for large blocks.
Is it possible to see whether such optimization does take place during the micro-benchmarks?